admin/app/controllers/solidus_admin/users_controller.rb

```diff
@@ -47,7 +78,26 @@ def set_user
     end
 
     def user_params
-      params.require(:user).permit(:user_id, permitted_user_attributes)
+      params.require(:user).permit(
+        :user_id,
+        permitted_user_attributes,
+        bill_address_attributes: permitted_address_attributes,
+        ship_address_attributes: permitted_address_attributes
+      )
+    end
+
+    # @note This method aims to fill the gap left by not having, or not knowing
+    # whether we will have the following on the current Spree.user_class.
+    #   accepts_nested_attributes_for :ship_address
+    #   accepts_nested_attributes_for :bill_address
+    # Not having these strips us of the ability to leverage a lot of the
+    # default Rails validations.
```

Comment:
Those are provided by the `UserAddressBook` module that is included in `UserMethods` which we expect to be included in all custom user classes.

https://github.com/solidusio/solidus/blob/3a92a45e94940641e9b4392c9a20d1241f651ce9/core/app/models/concerns/spree/user_address_book.rb#L41-L42

https://github.com/solidusio/solidus/blob/3a92a45e94940641e9b4392c9a20d1241f651ce9/core/app/models/concerns/spree/user_methods.rb#L9

I think it is fine if we assume this is the case. And even not, we can expect from people while upgrading to the new Admin to include this module in their custom user class.